.github/workflows/check.yaml

```diff
@@ -22,19 +26,30 @@ jobs:
       working-directory: ./src
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    name: Functional tests
```

Comment:
I'm curious about the reason behind discontinuing the use of func.yaml. Is there a specific reason why we can't simply modify the existing one or introduce a new version? What's the major challenges to upgrading the controller? Initially, I had anticipated simply adjusting the 'juju-channel' value to 3.4.

My concern is that we may modify check.yaml again when we eventually upgrade to version 3.5. Minimizing these changes would be preferable.

Therefore, I prefer to know the whole plan and announce to whole team. thanks!

Reply:
It was dou current version of those workflows does not support lxd-channel change and I hit the issue https://github.com/canonical/charmcraft/issues/1640. That issue was fixed.

My other reasons are:
- The reusable workflow for functional become too complex with too many parameters, so they are not readable what actually will be executed and how. If you open the pipeline, you see the lot of skipped steps, and you have no idea what actually was run.
- Any small change required us to change the workflow and then come back to the charm and if sometimes just created a new branch.
- We already have 4 branches, so it's not easy to change it on a single place.
- With potential Terraform use we can do it from one place, without that I would rather go and create 20 PRs with changing one parameter, than change it for 4 branches and then hope it works. 

This is an experiment and we can have discussion before merging any PR like this.

Reply:
As we internally discuss and the issue https://github.com/canonical/charmcraft/issues/1640 was fixed, I'll rewert this to using bootstack-actions again.